server/src/main/java/org/opensearch/rest/BaseRestHandler.java

```diff
@@ -122,9 +122,8 @@ public final void handleRequest(RestRequest request, RestChannel channel, NodeCl
             throw new IllegalArgumentException(unrecognized(request, unconsumedParams, candidateParams, "parameter"));
         }
 
-        if (request.hasContent() && request.isContentConsumed() == false) {
-            throw new IllegalArgumentException("request [" + request.method() + " " + request.path() + "] does not support having a body");
-        }
+        // ignore whether content is consumed
+        // https://github.com/opensearch-project/OpenSearch/issues/13011
```

Comment:
Nit: I'd rather remove the comment to keep the file cleaner (nothing to see here!) If folks are interested in new behavior they can look at the commit history